test/hotspot/gtest/runtime/test_os_linux.cpp

```diff
@@ -360,10 +360,10 @@ TEST_VM(os_linux, pretouch_thp_and_use_concurrent) {
   EXPECT_TRUE(os::commit_memory(heap, size, false));
 
   {
-    auto pretouch = [heap, size](Thread*, int) {
+    auto pretouch = [heap](Thread*, int) {
       os::pretouch_memory(heap, heap + size, os::vm_page_size());
     };
-    auto useMemory = [heap, size](Thread*, int) {
+    auto useMemory = [heap](Thread*, int) {
```

Comment:
Coming to this late, after the change has been integrated.

These lambdas should have been changed to use implicit reference capture
(`[&]`), per the style guide.

The change to the pretouch lambda actually makes things more difficult to
understand, since the outer `size` variable *is* referenced within the lambda
body.  This works because `size` is an integral constant initialized from a
constant expression, so does not need to be captured for the reference to
work. I'm guessing the clang warning had something to do with that, but it's
not mentioned in the JBS issue.  If implicit reference capture were used, the
referential transparency that comes with it would mean readers wouldn't need
to even know about that distinction.

I'll file an issue to change the capture lists.  I didn't find any other
explicit capture lists in HotSpot code.